### ISSUE TITLE
Implement Tessendorf waves

### ISSUE THREAD

**Issue by virtualritz** [Neopallium/bevy_water#23]
A C++ source that could easily be ported is e.g. available as part of the Houdini Ocean Toolkit here:

https://github.com/eloop/hot

**Comment by virtualritz** [comment]
> Do you know if it can run real-time?

Absolutely. I ported it to a Maya plugin over a decade ago and it ran realtime, CPU-only, then.

**Comment by Neopallium** [comment]
I did some looking into this.  I don't have the time to work on it right now, but it is interesting and I will try working on it in the future.

Some more useful links:

- https://github.com/blackencino/EncinoWaves - Seems to be based on/related to Tessendorf waves.
- https://projects.blender.org/blender/blender/src/branch/main/source/blender/blenkernel/intern/ocean.c - C code version for Blender.
